### ISSUE TITLE
PFAF is taking so long scraping and not finishing

### ISSUE THREAD

**Issue by RavanJAltaie** [openzim/zimit#257]
Hi, this recipe [here](https://farm.openzim.org/pipeline/dc09425d-c2d6-4f6a-9503-c0ae74236402) is taking very long time scraping and couple of times it reaches 89% and stops.
Please check

**Comment by benoit74** [comment]
Site is served/protected by Cloudflare.

After some pages, all pages are failing with the "Navigation timeout of 90000 ms exceeded" error. After even more time, all pages are returning an HTTP 500 error.

Clearly it looks like we are not welcomed to scrape this website.

As discussed live, once you begin to see that all pages are finishing with a "Navigation timeout of 90000 ms exceeded" errors, there is absolutely no reason to let the recipe continue, you must cancel it so save computation resources (the ZIM will miss many pages in any case, and we will decide to not publish it).

I do not have a real solution so far.

My recommendation is to attempt to set a "Page Extra Delay" of 10 seconds between pages and relaunch the recipe, to see if it helps avoid being blocked by Cloudflare. If this succeed, then we will experiment with this setting to find the minimal value to make the recipe as fast as possible.

**Comment by benoit74** [comment]
I cancelled the new task [97518383-67b7-45d2-be50-c4648bc3238b](https://farm.openzim.org/pipeline/97518383-67b7-45d2-be50-c4648bc3238b), it is still full of "Navigation timeout of 90000 ms exceeded".

Let's keep this open until we find a solution.
